### ISSUE TITLE
Bug: Default quantity unit consume should affect the unit of the quick consume amount

### ISSUE THREAD

**Issue by d-bieber** [grocy/grocy#2426]
At the moment the unit of the fields "energy", "quick consume amount" and "quick open amount" are all set by "Quantity unit stock". In my opinion it would make more sense if the unit of the consume amount correlates to the unit of "default quantity unit consume". 
Furthermore if I have a different unit for consumption (and an opened item), the amount in the stock overview should be in the consumption unit, too. Also the buttons should give the right amount and units for the opening/consumption.

I try to give an practical example: I have a pack chocolate which contains 5 pieces of chocolate. If I open it, I want to have 5 pieces. If I consume it, I don't want to consume in packs but in pieces (I know, I will consume all 5 of them in seconds, but that's not the point 😄). Right now I could only consume all 5 pieces of them and the mark as open button shows 5, even if I just open one pack.
<img width="715" alt="Unbenannt" src="https://github.com/grocy/grocy/assets/8823040/5af6cdbd-667a-448a-9a16-e50d474dd84b">

**Comment by berrnd** [comment]
The issue template explicitly mentions to keep it to one topic per request, since everything else is untrackable - would be cool if you could try to not ignore that next time. Furthermore, how are things you personally would like to have or simply basic usage questions supposed to be a Bug? I'm open to Feature Requests and answering questions, but allergic to calling wishes and things you don't know/understand yet a big bug.

&nbsp;
> At the moment the unit of the fields "energy", "quick consume amount" and "quick open amount" are all set by "Quantity unit stock". [...]

Duplicate of #2252 - please be so kind and do a little search beforehand (also that is mentioned on the request template).

&nbsp;
> I have a pack chocolate which contains 5 pieces of chocolate. If I open it, I want to have 5 pieces. If I consume it, I don't want to consume in packs but in pieces

Simply configure your product accordingly and everything will work. This is not the right place for usage questions ([README ref](https://github.com/grocy/grocy#questions--help--bug-reports--feature-requests)) and [persistent demo instances](https://www.reddit.com/r/grocy/comments/jwnk9b) are there to reproduce concrete examples, since nobody is able to look behind screenshots and there is always more around than visible there.

**Comment by d-bieber** [comment]
Hi, 
first, thanks for your fast answer. I'm sorry if my report was maybe not clear enough, so I try to give a better explanation, of why I think, this is a bug and why these two topics I mentioned are part of the same problem.
I'm sorry in advance if this is in fact a layer 8 problem and I overlooked something big, but maybe you can show me the right direction if this is the case.

While #2252 is similar I would not say that this is a duplicate. A different unit for the Energy field is a nice to have. Not a bug. But the unit of the "Quick consume amount" should correlate to the setting, which sets the unit of the consumption. Otherwise, there are two different default units of consumption, which is confusing for the user. And as these settings are represented on the stock overview page, my two topics originate from the same problem.

I'll try to describe again, what my goal is and how I tried to achieve it:
My goal is to represent items, where the content has a different unit (e.g. pieces). I don't want to consume the package but the pieces instead. The amount should be given in (closed) packs and (opened) pieces.
1. I created the units for this. In this case pack and piece.
2. I created the product "chocolate" and specify the QU conversions. (1 pack is 8 pieces).
3. I set "Default quantity unit consume" to Piece. I can't set the "quick consume amount" to 1 piece. Of course, I can set it to 0.125 Packs. This would be a quick fix but does not look good on the overview under amount. Below more. I save the product.
4. I purchase 1 Pack of chocolate.


If I set the amount to 0.125, the consume button rightfully says, "consume 1 piece of chocolate". The Button for "Mark open" is wrong and says "Mark 8 Pieces of chocolate as open". This is wrong, because I only open **1 Pack** not **8 Pieces**. The amount on the overview should now say, that there are 8 Pieces, but says "1 Pack, 1 opened". This gets especially problematic, if I consume 1 piece, because I now don't have **7 pieces**, but **0,875 Packs**.

So, my expected outcome would be:
- On the edit product page, the "Quick consume amount" would be given in Pieces, because I set Default quantity unit consume to Piece.
- On the Stock overview page, the button for "mark as open" should say "one Pack", not 8 Pieces.
- Also on stock overview, the amount column should reference the consume amount, because 0.875 Packs is not as clear as 7 Pieces. Especially if 1 Pack is not 8 Pieces, but for example 14 or 650 grams.


As you wished, I reproduced the problem in a private instance: https://test-pjajuhcavkjf64ckfk2mg.demo-prerelease.grocy.info/stockoverview

Again, I'm sorry, if there is something big, I missed, but I think there is at least some potential for finetuning.


Thank you for your work and greetings from cologne.
Dominik

**Comment by berrnd** [comment]
First, let's finally clarify that consuming something has definitely not anything to do with a energy unit - I don't get why mixing these completely unrelated things here is so important for you. #2252 will give you and everyone the freedom to define the energy value related to whatever unit you want, independent from any other, serving everyone's need. Remember: Grocy can't only follow your personal way of thinking. It also sounds you've discovered Grocy just recently, but it exists for years and is used by thousands of people. Based on that all, the current state of how stuff works in Grocy can't be so bad in general. And you still call your wish / usage question a Bug, cool.

Also, what's the point about insisting on doing each and every consume transaction on [/stockoverview](https://demo-prerelease.grocy.info/stockoverview)? It's just a page where you can get an overview of what's in stock. You can consume any amount related to any product related unit on [/consume](https://demo-prerelease.grocy.info/consume) (and [/stockentries](https://demo-prerelease.grocy.info/stockentries) is another related page).

So what you want is already possible, it only depends on definition / configuration of the product. Many ways to do that, I just created "Chocolate2" on the same persistent demo instance to showcase another approach. Everything works either way.

&nbsp;
> I can set it to 0.125 Packs. This would be a quick fix but does not look good on the overview under amount.

Grocy is about functionality and it works, not about beauty - which is obviously the only "problem" here.

&nbsp;
In conclusion (for the consume/open handling part) another duplicate of #2329 - please do little search beforehand next time. I'm going to lock this, since talking about the same thing over and over again is nothing else than frustrating. Maybe it turns out for you that Grocy is not the right tool to handle everything you personally need, which would be also just fine.

**Comment by berrnd** [comment]
Follow-up: I created #2427 to describe ideas for this topic in a general and decluttered way, feel free to add further ideas there if you think there is still something missing. But please remember that this all is still not a Bug and please also don't create another clutter there, thanks.
